### PR TITLE
add vending machines and code cleanup

### PR DESCRIPTION
**Pull request by wsor4035**
cleans up code and adds vending machines

### CODE REVIEW

**Review by BuckarooBanzay — APPROVED** [review]
Looks good, didn't test though

**Comment by OgelGames** [comment]
```
2021-11-29 13:29:40: ERROR[Main]: generateImage(): Could not load image "home_workshop_machines_drawer_small.png" while building texture; Creating a dummy image
```
Stuff should be tested before merging...

File: home_vending_machines/api.lua [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            local iname = itemstack:get_name()
+            local dpos = vector.add((vector.multiply(minetest.facedir_to_dir(node.param2), -1)), pos)
+
+            if registered_currency[iname] and registered_currency[iname] == 1 then
+                local item = def._vmachine.item
+                if type(item) == "table" then
+                    item = item[math.random(#item)]
+                end
+
+                minetest.spawn_item(dpos, item)
```
Comment (OgelGames) [inline home_vending_machines/api.lua:41]: Apparently `minetest.spawn_item` is an undocumented Lua-implemented version of `minetest.add_item`. 🤔

File: home_vending_machines/mod.conf [hunk 2]
```diff
@@ -0,0 +1 @@
+optional_depends = "screwdriver"
```
Comment (OgelGames) [inline home_vending_machines/mod.conf:1]: There shouldn't be quotes here.

File: home_workshop_misc/init.lua [hunk 3]
```diff
@@ -79,60 +79,13 @@ minetest.register_node("home_workshop_misc:beer_mug", {
 	sounds = default.node_sound_glass_defaults(),
 	selection_box = beer_cbox,
 	on_use = function(itemstack, user, pointed_thing)
-		if not minetest.is_creative_enabled_for(user:get_player_name()) then
+		if not minetest.is_creative_enabled(user:get_player_name()) then
```
Comment (OgelGames) [inline home_workshop_misc/init.lua:82]: `minetest.is_creative_enabled` is 5.3.0+ only. (mainly relevant for CDB minimum minetest version)

Comment (wsor4035) [reply home_workshop_misc/init.lua:82]: yup, and textures in dirs is 5.4 only

File: home_vending_machines/api.lua (hunk 1 above)
Comment (wsor4035) [reply home_vending_machines/api.lua:41]: yup

Comment (wsor4035) [reply home_vending_machines/api.lua:41]: probably should swap

**Comment by wsor4035** [comment]
> ```
> 2021-11-29 13:29:40: ERROR[Main]: generateImage(): Could not load image "home_workshop_machines_drawer_small.png" while building texture; Creating a dummy image
> ```
> 
> Stuff should be tested before merging...

for the record this is a legacy issue, not introduced by this PR when using this mod without homedecor

**Pull request merged**